_examples/custom_http_auth/main.go

```diff
@@ -0,0 +1,59 @@
+package main
+
+import (
+	"crypto/tls"
+	"fmt"
+	"net/http"
+	"os"
+	"time"
+
+	"gopkg.in/src-d/go-git.v4"
+	. "gopkg.in/src-d/go-git.v4/_examples"
+	"gopkg.in/src-d/go-git.v4/plumbing/transport/client"
+	githttp "gopkg.in/src-d/go-git.v4/plumbing/transport/http"
+	"gopkg.in/src-d/go-git.v4/storage/memory"
+)
+
+// Here is an example to configure http client according to our own needs.
+func main() {
+	CheckArgs("<url>", "<username>", "<password>")
```

Comment:
To get the tests for the example code to pass, you'll need to set some arguments in the test file: https://github.com/src-d/go-git/blob/master/_examples/common_test.go#L17

```
=== RUN   TestExamples/custom_http_auth
Usage: /tmp/go-build169800819/command-line-arguments/_obj/exe/main <url> <username> <password>
exit status 1
```